### PR TITLE
Fix/exogenous variables prophet

### PR DESCRIPTION
**Pull request by wilfreddesert**
I spent a few hours trying to understand why adding exogenous variables couldn't work properly. 
After a few small fixes, now I think it works properly at least for Facebook Prophet. 

You can take a look at the diff, but please also note that there are some changes caused by running `black`. I also edited some tests just for the sake of convenience and added a test for exogenous variables. 

If some of the changes are not okay, please let me know :)

### CODE REVIEW

File: hts/core/regressor.py [hunk 1]
```diff
@@ -364,10 +365,14 @@ def _revise(self, steps_ahead: int = 1) -> pandas.DataFrame:
 
     def _get_predict_index(self, steps_ahead=1) -> Any:
 
-        freq = getattr(self.nodes.item.index, "freq", 1)
-        start = self.nodes.item.index[-1] + freq
-        end = self.nodes.item.index[-1] + (steps_ahead * freq)
-
-        future = pandas.date_range(freq=freq, start=start, end=end)
+        freq = getattr(self.nodes.item.index, "freq", 1) or 1
+        try:
```
Comment (carlomazzaferro) [inline hts/core/regressor.py:369]: Good point here, looks good

File: tests/conftest.py [hunk 2]
```diff
@@ -113,21 +113,28 @@ def hierarchical_mv_data():
 
 
 @pytest.fixture
-def mv_tree(hierarchical_mv_data):
+def mv_tree_empty():
```
Comment (carlomazzaferro) [inline tests/conftest.py:116]: Nice refactor!

**Comment by carlomazzaferro** [comment]
This is looking quite good -- just out of curiosity, how are you running the linting step? Surprisingly, despite the changes the linting step still works in CI

**Comment by wilfreddesert** [comment]
> This is looking quite good -- just out of curiosity, how are you running the linting step? Surprisingly, despite the changes the linting step still works in CI

I was surprised by that as well. I followed the setup steps exactly so `black` should work in the same manner. Nonetheless, it reformatted some parts of the older code, but then all the checks passed :)

**Comment by carlomazzaferro** [comment]
No objections, will approve and merge! Thanks for the hard and quick work :)

**Pull request merged**